swarm/api/http/roundtripper.go

```diff
@@ -43,7 +43,8 @@ func (self *RoundTripper) RoundTrip(req *http.Request) (resp *http.Response, err
 	if len(host) == 0 {
 		host = "localhost"
 	}
-	url := fmt.Sprintf("http://%s:%s/%s/%s", host, self.Port, req.URL.Host, req.URL.Path)
+	url := fmt.Sprintf("http://%s:%s/%s:/%s/%s", host, self.Port, req.Proto, req.URL.Host, req.URL.Path)
 	glog.V(logger.Info).Infof("[BZZ] roundtripper: proxying request '%s' to '%s'", req.RequestURI, url)
-	return http.Get(url)
+	reqProxy, _ := http.NewRequest(req.Method, url, req.Body)
```

Comment:
why are we sure there cannot be an error here?
here is the code for http.NewRequest : https://golang.org/src/net/http/request.go?s=19631:19703#L588 
Notably if method is not valid or the url does not parse the error is non-nil.
I think we need to guard for that.